js/noms/src/browser/bytes.js

```diff
@@ -20,11 +18,14 @@ export function fromValues(values: number[]): Uint8Array {
 }
 
 export function fromString(s: string): Uint8Array {
-  return encoder.encode(s);
+  const size = byteLength(s);
+  const buf = alloc(size);
```

Comment:
One other option would be to have a scratch buffer here so that we do not have to compute the `byteLength` in advance.